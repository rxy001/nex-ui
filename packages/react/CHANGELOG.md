# @nex-ui/react

## 0.0.15

### Patch Changes

- fdd4483: Release
- Updated dependencies [fdd4483]
  - @nex-ui/styled@0.0.15
  - @nex-ui/system@0.0.15
  - @nex-ui/hooks@0.0.15
  - @nex-ui/icons@0.0.15
  - @nex-ui/utils@0.0.15

## 0.0.14

### Patch Changes

- ca25983: Release
- Updated dependencies [ca25983]
  - @nex-ui/styled@0.0.14
  - @nex-ui/system@0.0.14
  - @nex-ui/hooks@0.0.14
  - @nex-ui/icons@0.0.14
  - @nex-ui/utils@0.0.14

## 0.0.13

### Patch Changes

- 1b97788: Release
- Updated dependencies [1b97788]
  - @nex-ui/styled@0.0.13
  - @nex-ui/system@0.0.13
  - @nex-ui/hooks@0.0.13
  - @nex-ui/icons@0.0.13
  - @nex-ui/utils@0.0.13

## 0.0.12

### Patch Changes

- 664eb21: Test

## 0.0.11

### Patch Changes

- 259356e: Test
- Updated dependencies [259356e]
  - @nex-ui/styled@0.0.11
  - @nex-ui/system@0.0.11
  - @nex-ui/hooks@0.0.11
  - @nex-ui/icons@0.0.11
  - @nex-ui/utils@0.0.10

## 0.0.10

### Patch Changes

- 5a229bd: release
- Updated dependencies [5a229bd]
  - @nex-ui/styled@0.0.10
  - @nex-ui/system@0.0.10
  - @nex-ui/icons@0.0.10

## 0.0.9

### Patch Changes

- 4a5c359: release
- Updated dependencies [4a5c359]
  - @nex-ui/styled@0.0.9
  - @nex-ui/system@0.0.9
  - @nex-ui/hooks@0.0.9
  - @nex-ui/icons@0.0.9
  - @nex-ui/utils@0.0.9

## 0.0.8

### Patch Changes

- c8c1956: release
- Updated dependencies [c8c1956]
  - @nex-ui/styled@0.0.8
  - @nex-ui/system@0.0.8
  - @nex-ui/hooks@0.0.8
  - @nex-ui/icons@0.0.8
  - @nex-ui/utils@0.0.8

## 0.0.7

### Patch Changes

- 2618e1a: release
- Updated dependencies [2618e1a]
  - @nex-ui/styled@0.0.7
  - @nex-ui/system@0.0.7
  - @nex-ui/hooks@0.0.7
  - @nex-ui/icons@0.0.7
  - @nex-ui/utils@0.0.7

## 0.0.6

### Patch Changes

- 87191c1: release
- Updated dependencies [87191c1]
  - @nex-ui/styled@0.0.6
  - @nex-ui/system@0.0.6
  - @nex-ui/hooks@0.0.6
  - @nex-ui/icons@0.0.6
  - @nex-ui/utils@0.0.6

## 0.0.5

### Patch Changes

- 7631577: release
- Updated dependencies [7631577]
  - @nex-ui/styled@0.0.5
  - @nex-ui/system@0.0.5
  - @nex-ui/hooks@0.0.5
  - @nex-ui/icons@0.0.5
  - @nex-ui/utils@0.0.5

## 0.0.4

### Patch Changes

- 70cb478: v0.0.4
- Updated dependencies [70cb478]
  - @nex-ui/styled@0.0.4
  - @nex-ui/system@0.0.4
  - @nex-ui/hooks@0.0.4
  - @nex-ui/icons@0.0.4
  - @nex-ui/utils@0.0.4
