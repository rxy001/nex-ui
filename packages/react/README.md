<div style="background:white;">
  <p align="center">
    <img src='../../assets/logo.png' style="width: 150px; height: 150px;" />
  </p>
  <h1 align="center">Nex UI</h1>
  <p align="center">ðŸŽ‰ A beautiful, modern, and reliable React component library.</p>
  <p align="center"><b>Theme Customizable, Light & Dark UI, Unique DX, Written In TypeScript</b></p>
</div>

## Documentation

Visit https://nex-ui-docs.vercel.app to view the full documentation.

## Getting Started

Visit https://nex-ui-docs.vercel.app/docs/getting-started to get started with Nex UI.

## Storybook

Visit https://nex-ui-storybook.vercel.app to view the storybook for all components.

## Contributing

Contributions are always welcome!

See [CONTRIBUTING.md](/CONTRIBUTING.md) for ways to get started.

Please adhere to this project's [CODE_OF_CONDUCT](/CODE_OF_CONDUCT.md).

## License

[MIT](https://choosealicense.com/licenses/mit/)
