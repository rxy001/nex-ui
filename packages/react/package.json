{
  "name": "@nex-ui/react",
  "version": "0.0.12",
  "author": "X1ng Yu",
  "description": "ðŸŽ‰ A beautiful, modern, and reliable React component library.",
  "homepage": "https://nex-ui-docs.vercel.app/",
  "private": false,
  "keywords": [
    "react",
    "react-component",
    "nui",
    "nex-ui"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/rxy001/nex-ui.git",
    "directory": "packages/react"
  },
  "license": "MIT",
  "scripts": {
    "build": "tsx ../../scripts/build.ts --dts",
    "build:fast": "tsx ../../scripts/build.ts",
    "typecheck": "tsc --noEmit",
    "prepack": "clean-package",
    "postpack": "clean-package restore"
  },
  "type": "module",
  "files": [
    "dist"
  ],
  "main": "src/index.ts",
  "devDependencies": {
    "@storybook/react": "8.6.12"
  },
  "dependencies": {
    "@nex-ui/hooks": "workspace:*",
    "@nex-ui/icons": "workspace:*",
    "@nex-ui/styled": "workspace:*",
    "@nex-ui/system": "workspace:*",
    "@nex-ui/utils": "workspace:*",
    "clsx": "2.1.1"
  },
  "peerDependencies": {
    "@emotion/react": ">=11",
    "motion": "^12.7.2",
    "react": ">=19",
    "react-dom": ">=19"
  },
  "sideEffects": false,
  "clean-package": "../../clean-package.config.json"
}
