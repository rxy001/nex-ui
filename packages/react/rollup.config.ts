import path from 'node:path'
import { fileURLToPath } from 'node:url'
import { defineConfig } from 'rollup'
import typescript from '@rollup/plugin-typescript'
import del from 'rollup-plugin-delete'
import dts from 'rollup-plugin-dts'
import { nexUIRollupPlugin } from '@nex-ui/plugins'

const dirname = fileURLToPath(new URL('./', import.meta.url))

export default defineConfig([
  {
    plugins: [
      del({
        targets: './dist/*',
      }),
      typescript({
        tsconfig: path.resolve(dirname, 'tsconfig.json'),
      }),
      nexUIRollupPlugin(),
    ],
    input: './src/index.ts',
    external: [
      'react',
      'react-dom',
      'react/jsx-runtime',
      'react-dom/client',
      '@nex-ui/utils',
      '@nex-ui/css-system',
      'classnames',
      'framer-motion',
      '@iconify/react',
      '@vanilla-extract/dynamic',
      '@vanilla-extract/css',
      '@vanilla-extract/css/functionSerializer',
    ],
    output: [
      {
        preserveModules: true,
        format: 'es',
        dir: './dist/es',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
      {
        preserveModules: true,
        format: 'cjs',
        dir: './dist/lib',
        assetFileNames({ name }) {
          return name?.replace(/^src\//, '') ?? ''
        },
      },
    ],
  },
  {
    plugins: [dts()],
    input: './src/index.ts',
    output: {
      dir: './dist/types',
      format: 'es',
      preserveModules: true,
    },
  },
])
