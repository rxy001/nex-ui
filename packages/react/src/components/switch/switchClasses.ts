import { generateUtilityClasses } from '../utils'

// 用于单元测试
export const switchClasses = generateUtilityClasses('nui-switch', [
  // ---------root---------
  'root',

  // "blue" | "gray" | "pink" | "purple" | "cyan" | "yellow" | "orange" | "red" | "green"
  'color-gray',
  'color-green',
  'color-red',
  'color-yellow',
  'color-blue',
  'color-pink',
  'color-cyan',
  'color-purple',
  'color-orange',

  // "sm" | "md" | "lg"
  'size-sm',
  'size-md',
  'size-lg',

  'disabled',
  'checked',
  // ---------root---------

  'input',
  'track',
  'thumb',
  'start-icon',
  'end-icon',
  'label',
])
