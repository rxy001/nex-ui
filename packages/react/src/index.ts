import './types/declare'

export * from './components'
export * from './theme'

export type { Theme } from './types/theme'
export type { ComponentsTheme } from './types/componentsTheme'

export { useColorScheme } from '@nex-ui/system'
export { nex, styled } from '@nex-ui/styled'
