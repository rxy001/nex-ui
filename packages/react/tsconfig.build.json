{
  "extends": "../../tsconfig.json",
  "include": ["./src"],
  "exclude": ["./src/**/__tests__/**/*", "./src/**/__stories__/**/*"]
}
