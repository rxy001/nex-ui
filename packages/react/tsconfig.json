{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "~/*": ["../../*"],
      "@theme": ["./src/theme"]
    }
  },
  "include": ["./src", "./rollup.config.ts"]
}
