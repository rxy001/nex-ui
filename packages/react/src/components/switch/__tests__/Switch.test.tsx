import { createRef, useState } from 'react'
import { act, fireEvent } from '@testing-library/react'
import {
  testComponentStability,
  renderWithNexUIProvider,
  testRootClassName,
  testColorClasses,
  testSizeClasses,
} from '~/tests/shared'
import { Switch } from '../index'
import { switchClasses } from '../switchClasses'

describe('Switch', () => {
  testComponentStability(<Switch />)

  testRootClassName(<Switch className='test-class' />, 'test-class')

  testColorClasses(<Switch />, switchClasses)

  testSizeClasses(<Switch />, switchClasses)

  it('should render with default props', () => {
    const { container, getByRole } = renderWithNexUIProvider(<Switch />)

    const root = container.firstElementChild
    const input = getByRole('switch')

    expect(root).toMatchSnapshot()
    expect(root).toHaveClass(switchClasses.root)
    expect(root).toHaveClass(switchClasses['color-blue'])
    expect(root).toHaveClass(switchClasses['size-md'])
    expect(input).toHaveClass(switchClasses.input)

    expect(root).not.toHaveClass(switchClasses.disabled)
    expect(root).not.toHaveClass(switchClasses.checked)
    expect(root).not.toHaveClass(switchClasses['color-cyan'])
    expect(root).not.toHaveClass(switchClasses['color-green'])
    expect(root).not.toHaveClass(switchClasses['color-gray'])
    expect(root).not.toHaveClass(switchClasses['color-orange'])
    expect(root).not.toHaveClass(switchClasses['color-pink'])
    expect(root).not.toHaveClass(switchClasses['color-purple'])
    expect(root).not.toHaveClass(switchClasses['color-red'])
    expect(root).not.toHaveClass(switchClasses['color-yellow'])
    expect(root).not.toHaveClass(switchClasses['size-sm'])
    expect(root).not.toHaveClass(switchClasses['size-lg'])
  })

  it("should forward ref to Switch's input element", () => {
    const ref = createRef<HTMLInputElement>()
    const { getByRole } = renderWithNexUIProvider(<Switch ref={ref} />)
    expect(getByRole('switch')).toBe(ref.current)
  })

  it('should forward slotProps to root, track, thumb, startIcon, endIcon and label slots', () => {
    const slotProps = {
      root: {
        className: 'test-root-class',
      },
      track: {
        className: 'test-track-class',
      },
      thumb: {
        className: 'test-thumb-class',
      },
      startIcon: {
        className: 'test-start-icon-class',
      },
      endIcon: {
        className: 'test-end-icon-class',
      },
      label: {
        className: 'test-label-class',
      },
    }

    const { container } = renderWithNexUIProvider(
      <Switch
        slotProps={slotProps}
        startIcon={<span>Start</span>}
        endIcon={<span>End</span>}
      >
        Switch
      </Switch>,
    )

    const root = container.firstElementChild
    const track = container.querySelector(`.${switchClasses.track}`)
    const thumb = container.querySelector(`.${switchClasses.thumb}`)
    const startIcon = container.querySelector(`.${switchClasses['start-icon']}`)
    const endIcon = container.querySelector(`.${switchClasses['end-icon']}`)
    const label = container.querySelector(`.${switchClasses.label}`)

    expect(root).toHaveClass(slotProps.root.className)
    expect(track).toHaveClass(slotProps.track.className)
    expect(thumb).toHaveClass(slotProps.thumb.className)
    expect(startIcon).toHaveClass(slotProps.startIcon.className)
    expect(endIcon).toHaveClass(slotProps.endIcon.className)
    expect(label).toHaveClass(slotProps.label.className)
  })

  it('should forward classes to root, track, thumb, startIcon, endIcon, label and input slots', () => {
    const classes = {
      root: 'test-root-class',
      track: 'test-track-class',
      thumb: 'test-thumb-class',
      startIcon: 'test-start-icon-class',
      endIcon: 'test-end-icon-class',
      label: 'test-label-class',
    }
    const { container } = renderWithNexUIProvider(
      <Switch
        classes={classes}
        startIcon={<span>Start</span>}
        endIcon={<span>End</span>}
      >
        Switch
      </Switch>,
    )

    const root = container.firstElementChild
    const track = container.querySelector(`.${switchClasses.track}`)
    const thumb = container.querySelector(`.${switchClasses.thumb}`)
    const startIcon = container.querySelector(`.${switchClasses['start-icon']}`)
    const endIcon = container.querySelector(`.${switchClasses['end-icon']}`)
    const label = container.querySelector(`.${switchClasses.label}`)

    expect(root).toHaveClass(classes.root)
    expect(track).toHaveClass(classes.track)
    expect(thumb).toHaveClass(classes.thumb)
    expect(startIcon).toHaveClass(classes.startIcon)
    expect(endIcon).toHaveClass(classes.endIcon)
    expect(label).toHaveClass(classes.label)
  })

  it('should render label with children', () => {
    const { getByText } = renderWithNexUIProvider(<Switch>Switch</Switch>)

    expect(getByText('Switch')).toBeInTheDocument()
  })

  it('should support defaultChecked prop', async () => {
    const { container, getByRole } = renderWithNexUIProvider(
      <Switch defaultChecked />,
    )

    const root = container.firstElementChild
    const input = getByRole('switch')

    expect(root).toHaveClass(switchClasses.checked)
    expect(input).toBeChecked()

    await act(async () => {
      fireEvent.click(input)
    })

    expect(root).not.toHaveClass(switchClasses.checked)
    expect(input).not.toBeChecked()

    await act(async () => {
      fireEvent.click(input)
    })

    expect(root).toHaveClass(switchClasses.checked)
    expect(input).toBeChecked()
  })

  it('should switch between checked and unchecked states when clicked', async () => {
    const { container, getByRole } = renderWithNexUIProvider(<Switch />)

    const root = container.firstElementChild!
    const input = getByRole('switch')

    expect(root).not.toHaveClass(switchClasses.checked)
    expect(input).not.toBeChecked()

    await act(async () => {
      fireEvent.click(root)
    })

    expect(root).toHaveClass(switchClasses.checked)
    expect(input).toBeChecked()

    await act(async () => {
      fireEvent.click(root)
    })

    expect(root).not.toHaveClass(switchClasses.checked)
    expect(input).not.toBeChecked()
  })

  it('should be controlled by checked prop', async () => {
    function ControlledSwitch() {
      const [checked, setChecked] = useState<boolean>(false)

      return (
        <Switch
          checked={checked}
          onCheckedChange={(newChecked) => {
            setChecked(newChecked)
          }}
        />
      )
    }

    const { getByRole, container } = renderWithNexUIProvider(
      <ControlledSwitch />,
    )

    const input = getByRole('switch')
    const root = container.firstElementChild

    expect(input).not.toBeChecked()
    expect(root).not.toHaveClass(switchClasses.checked)

    await act(async () => {
      fireEvent.click(input)
    })

    expect(input).toBeChecked()
    expect(root).toHaveClass(switchClasses.checked)

    await act(async () => {
      fireEvent.click(input)
    })

    expect(input).not.toBeChecked()
    expect(root).not.toHaveClass(switchClasses.checked)
  })

  it('should call onCheckedChange when checked state changes', async () => {
    const onCheckedChange = jest.fn()
    const { getByRole } = renderWithNexUIProvider(
      <Switch onCheckedChange={onCheckedChange} />,
    )

    const input = getByRole('switch')

    await act(async () => {
      fireEvent.click(input)
    })

    expect(onCheckedChange).toHaveBeenCalledWith(true)

    await act(async () => {
      fireEvent.click(input)
    })

    expect(onCheckedChange).toHaveBeenCalledWith(false)
  })

  it('should disable switch when disabled prop is true', () => {
    const { getByRole, container } = renderWithNexUIProvider(
      <Switch disabled />,
    )

    const root = container.firstElementChild
    const input = getByRole('switch')

    expect(input).toBeDisabled()
    expect(root).toHaveClass(switchClasses.disabled)
    expect(root).toHaveStyle('pointer-events: none')
  })

  it('should render startIcon and endIcon', () => {
    const { container } = renderWithNexUIProvider(
      <Switch startIcon={<span>Start</span>} endIcon={<span>End</span>} />,
    )

    const startIcon = container.querySelector(`.${switchClasses['start-icon']}`)
    const endIcon = container.querySelector(`.${switchClasses['end-icon']}`)

    expect(startIcon).toBeInTheDocument()
    expect(endIcon).toBeInTheDocument()
    expect(startIcon?.textContent).toBe('Start')
    expect(endIcon?.textContent).toBe('End')
  })

  it('should render custom thumbIcon', () => {
    const { getByTestId, container, rerender } = renderWithNexUIProvider(
      <Switch thumbIcon={<span data-testid='thumb-icon'>Thumb Icon</span>} />,
    )

    const thumbIcon = getByTestId('thumb-icon')
    expect(thumbIcon).toBeInTheDocument()

    const wrapper = container.querySelector(`.${switchClasses.thumb}`)
    expect(wrapper).toContainElement(thumbIcon)

    const mockFn = jest.fn(() => (
      <span data-testid='thumb-icon'>Thumb Icon</span>
    ))

    rerender(<Switch thumbIcon={mockFn} />)
    const newThumbIcon = getByTestId('thumb-icon')
    expect(newThumbIcon).toBeInTheDocument()
    expect(wrapper).toContainElement(newThumbIcon)
    expect(mockFn).toHaveBeenCalledWith({
      checked: false,
      disabled: false,
      size: 'md',
      color: 'blue',
      defaultChecked: false,
      thumbIcon: mockFn,
    })
  })

  it('should have data-focus-visible attribute when focused', async () => {
    const { getByRole, user } = renderWithNexUIProvider(
      <Switch>Checkbox</Switch>,
    )

    const switchElement = getByRole('switch')

    expect(switchElement).not.toHaveAttribute('data-focus-visible')

    await user.tab()
    expect(document.activeElement).toBe(switchElement)
    expect(switchElement).toHaveAttribute('data-focus-visible', 'true')
  })

  describe('Accessibility', () => {
    it('should have type="checkbox" by default', () => {
      const { getByRole } = renderWithNexUIProvider(<Switch />)
      const input = getByRole('switch')
      expect(input).toHaveAttribute('type', 'checkbox')
    })

    it('should have role="switch" by default', () => {
      const { container } = renderWithNexUIProvider(<Switch />)
      const input = container.querySelector(`.${switchClasses.input}`)
      expect(input).toHaveRole('switch')
    })

    it('should have tabIndex=0 by default', () => {
      const { getByRole } = renderWithNexUIProvider(<Switch />)
      const input = getByRole('switch')
      expect(input).toHaveAttribute('tabIndex', '0')
    })

    it('should apply checked attribute when checked', () => {
      const { getByRole } = renderWithNexUIProvider(<Switch checked />)
      const input = getByRole('switch')
      expect(input).toHaveAttribute('checked', '')
      expect(input).not.toHaveAttribute('aria-checked', 'true')
    })

    it('should apply disabled attribute when disabled', async () => {
      const { getByRole } = await renderWithNexUIProvider(<Switch disabled />)
      const switchElement = getByRole('switch')

      expect(switchElement).toBeDisabled()
      expect(switchElement).not.toHaveAttribute('aria-disabled')
    })

    it('should apply aria-label when children is a string', () => {
      const { getByRole } = renderWithNexUIProvider(<Switch>Switch</Switch>)
      const input = getByRole('switch')
      expect(input).toHaveAttribute('aria-label', 'Switch')
    })

    it('should apply aria-labelledby when aria-labelledby prop is provided', () => {
      const { getByRole } = renderWithNexUIProvider(
        <Switch aria-labelledby='label-id'>
          <span id='label-id'>Switch</span>
        </Switch>,
      )
      const input = getByRole('switch')
      expect(input).toHaveAttribute('aria-labelledby', 'label-id')
    })

    it('should apply aria-labelledby="label-id" when children is a string', () => {
      const { getByRole, container } = renderWithNexUIProvider(
        <Switch>Switch</Switch>,
      )
      const input = getByRole('switch')
      const label = container.querySelector(`.${switchClasses.label}`)

      expect(input).toHaveAttribute('aria-labelledby', label!.id)
    })

    it('should apply aria-label when aria-label prop is provided', () => {
      const { getByRole } = renderWithNexUIProvider(
        <Switch aria-label='Custom Switch'>Switch</Switch>,
      )
      const input = getByRole('switch')
      expect(input).toHaveAttribute('aria-label', 'Custom Switch')
    })

    it('should apply aria-checked="true" to non-input elements when checked', () => {
      const { getByRole } = renderWithNexUIProvider(<Switch checked as='div' />)
      const input = getByRole('switch')

      expect(input).toHaveAttribute('aria-checked', 'true')
      expect(input).not.toHaveAttribute('checked')
    })

    it('should apply aria-checked="false" to non-input elements when unchecked', () => {
      const { getByRole } = renderWithNexUIProvider(<Switch as='div' />)
      const input = getByRole('switch')

      expect(input).toHaveAttribute('aria-checked', 'false')
      expect(input).not.toHaveAttribute('checked')
    })

    it('should apply aria-disabled="true" to non-input elements when disabled', () => {
      const { getByRole } = renderWithNexUIProvider(
        <Switch disabled as='div' />,
      )
      const input = getByRole('switch')

      expect(input).toHaveAttribute('aria-disabled', 'true')
      expect(input).not.toBeDisabled()
    })

    it('should check non-input elements when clicked', () => {
      const { getByRole } = renderWithNexUIProvider(<Switch as='div' />)
      const input = getByRole('switch')

      expect(input).toHaveAttribute('aria-checked', 'false')

      fireEvent.click(input)

      expect(input).toHaveAttribute('aria-checked', 'true')
    })

    it('should activate non-input elements when space is pressed', async () => {
      const onChange = jest.fn()
      const { user, getByRole } = renderWithNexUIProvider(
        <Switch as='span' onCheckedChange={onChange}>
          Focusable Span
        </Switch>,
      )

      const span = getByRole('switch')
      await user.tab()
      expect(document.activeElement).toBe(span)

      await user.keyboard(' ')
      expect(span).toHaveAttribute('aria-checked', 'true')
      expect(onChange).toHaveBeenCalledWith(true)

      await user.keyboard(' ')
      expect(span).toHaveAttribute('aria-checked', 'false')
      expect(onChange).toHaveBeenCalledWith(false)
    })
  })
})
