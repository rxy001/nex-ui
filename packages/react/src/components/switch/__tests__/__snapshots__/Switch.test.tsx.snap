// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render with default props 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  position: relative;
  height: var(--nui-sizes-8);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  -webkit-tap-highlight-color: transparent;
}

.emotion-1 {
  margin: 0;
  padding: 0;
  border: none;
  background: none;
  outline: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  border-radius: 0;
  box-shadow: none;
  box-sizing: border-box;
  position: absolute;
  inset: 0;
  opacity: 0;
  z-index: 1;
  cursor: inherit;
}

.emotion-1[type="search"]::-webkit-search-decoration {
  -webkit-appearance: none;
}

.emotion-1[type="search"]::-webkit-search-cancel-button {
  -webkit-appearance: none;
}

.emotion-1:is(:-webkit-autofill, :autofill) {
  background-color: transparent!important;
  -webkit-transition: background-color 50000s ease-in-out 0s;
  transition: background-color 50000s ease-in-out 0s;
}

.emotion-1[data-focus-visible=true]:not(:disabled, [data-disabled=true])+* {
  outline: var(--nui-borders-md) var(--nui-colors-blue-primary);
  outline-offset: var(--nui-spaces-0-5);
}

.emotion-2 {
  background-color: var(--nui-colors-gray-200);
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: var(--nui-radii-full);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  -webkit-transition: var(--nui-transitions-colors);
  transition: var(--nui-transitions-colors);
  overflow: hidden;
  position: relative;
  -webkit-padding-start: var(--nui-spaces-1);
  padding-inline-start: var(--nui-spaces-1);
  -webkit-padding-end: var(--nui-spaces-1);
  padding-inline-end: var(--nui-spaces-1);
  width: var(--nui-sizes-12);
  height: var(--nui-sizes-7);
  font-size: var(--nui-font-sizes-lg);
}

[data-nui-color-scheme="dark"] .emotion-2 {
  background-color: var(--nui-colors-gray-500);
}

.emotion-3 {
  background-color: var(--nui-colors-white);
  border-radius: inherit;
  -webkit-transition: var(--nui-transitions-margin);
  transition: var(--nui-transitions-margin);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: var(--nui-sizes-5);
  height: var(--nui-sizes-5);
}

<label
  class="nui-switch-root emotion-0"
  data-checked="false"
  data-color="blue"
  data-disabled="false"
  data-size="md"
>
  <input
    class="nui-switch-input emotion-1"
    role="switch"
    tabindex="0"
    type="checkbox"
  />
  <span
    class="nui-switch-track emotion-2"
  >
    <span
      class="nui-switch-thumb emotion-3"
    />
  </span>
</label>
`;
